bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/DefaultPerChannelBookieClientPool.java

```diff
@@ -63,12 +64,20 @@
         this.address = address;
         this.conf = conf;
 
-        this.shFactory = SecurityProviderFactoryFactory
-                .getSecurityProviderFactory(conf.getTLSProviderFactoryClass());
+        this.shFactory = SecurityProviderFactoryFactory.getSecurityProviderFactory(conf.getTLSProviderFactoryClass());
 
         this.clients = new PerChannelBookieClient[coreSize];
         for (int i = 0; i < coreSize; i++) {
-            this.clients[i] = factory.create(address, this, shFactory);
+            this.clients[i] = factory.create(address, this, shFactory, false);
+        }
+
+        if (conf.getUseV2WireProtocol()) {
+            this.clientsV3 = new PerChannelBookieClient[coreSize];
+            for (int i = 0; i < coreSize; i++) {
+                this.clientsV3[i] = factory.create(address, this, shFactory, true);
+            }
+        } else {
+            this.clientsV3 = null;
```

Comment:
assign `clients` to `clientsV3` here. try to avoid having a field to be null.

Reply:
ok